app/controllers/planning.rb

```diff
@@ -0,0 +1,107 @@
+get '/mission/new' do
+  #this should only provide currently avaliable ppl,
+  #off-line until the 'prevent double booking' validation is added
+  @personnel = Soldier.all
+  #as w/ ppl
+  @trucks = Truck.all
+  @trailers = Trailer.all
+  erb :'planning/new_mission'
+end
+
+
+post '/mission' do
+ 
+  Mission.transaction do
+
+    mission = Mission.new 
+
+    mission.step_off      ="#{params[:mission][:step_off_date]+" "+params[:mission][:step_off_time]+":00"}"
+    mission.return        ="#{params[:mission][:return_date]+" "+params[:mission][:return_time]+":00"}"
+    mission.name          = params[:mission][:name]
+    mission.unit_serviced = params[:mission][:unit_serviced]
+
+
+    # this is clearly a method; needs refactoring
+    dispatches = []
+    params[:truck].each do |dispatch_params|
+      truck = Truck.find_by!(name: dispatch_params[:truck_name])
```

Comment:
we should never ever ever ever (times ten) use the bang methods in our controllers. the bang methods blow up on us (which is great as developers), but not when a user is interacting with our application. thus, we as developers use the bang methods in our seed files and testing files - when we absolutely need to know if something is created or not. instead here, we should control for something not being found in a different way (perhaps a conditional)